src/Features/Core/Portable/Diagnostics/DiagnosticsUpdatedArgs.cs

```diff
@@ -14,7 +14,7 @@ internal sealed class DiagnosticsUpdatedArgs : UpdatedEventArgs
     public DiagnosticsUpdatedKind Kind { get; }
     public Solution? Solution { get; }
 
-    private readonly ImmutableArray<DiagnosticData> _diagnostics;
+    public readonly ImmutableArray<DiagnosticData> Diagnostics;
```

Comment:
This being a field is odd. Guessing you meant this to be:
```suggestion
    public ImmutableArray<DiagnosticData> Diagnostics { get; }
```

Reply:
For internal fields I do not mind readonly fields at all b:-)

Reply:
have switched them all to readonly fields.